packages/material-ui/src/Popper/Popper.js

```diff
@@ -13,7 +13,7 @@ import ownerWindow from '../utils/ownerWindow';
  * @param {string} placement
  */
 function flipPlacement(placement) {
-  const direction = (typeof window !== 'undefined' && document.body.getAttribute('dir')) || 'ltr';
+  const direction = (typeof document !== 'undefined' && document.body.getAttribute('dir')) || 'ltr';
```

Comment:
On a different note: Shouldn't we read from the theme here anyway. This ensures less issues during hydration and allows nested themes.

Reply:
I don't fully remember, it was either a tradeoff to minimize the size of the module (no default theme import) or to fix an update issue. It's likely not the only solution that we could have used. I guess we could use the theme context provider without the default theme instead. 

What's the use case for not using RTL for the whole document? On a side note, RTL usage is relatively low, maybe 5%.

Reply:
We support nesting themes therefore we support switching rtl (e.g. for sliders). We don't need to pull in the default theme if we have ltr by default.

Reply:
> We support nesting themes therefore we support switching rtl (e.g. for sliders)

I'm sorry, I don't understand the causality. Why does theme nesting support imply RTL nesting support? I think that, if it's simpler, we could clarify our position on this point and officials say we don't.

I have looked at the history of the code, the current tradeoff was related to bundle size, not update issue: https://github.com/mui-org/material-ui/pull/13804/files#diff-207f4ebb9e26ba4aa6c8a8148ee2b103R8. This was changed a year ago. What do you think of using `useTheme` from @material-ui/styles?

Reply:
> Why does theme nesting support imply RTL nesting support? 

If we support nesting theme and we merge every property when nesting we do support nesting every property.

> What do you think of using useTheme() from @material-ui/styles instead of reading the DOM?

Yep that's what I was trying to say by not needing to pull in the default theme.

Reply:
What's more important is hydration mismatch though.

Reply:
Ah, the placement leaks in `const childProps = { placement };` to userland. Ok so yeah useTheme seems definitely better.